src/webots/core/WbHttpReply.cpp

```diff
@@ -41,25 +41,30 @@ QByteArray WbHttpReply::forgeHTMLReply(const QString &htmlContent) {
   return reply;
 }
 
-QByteArray WbHttpReply::forgeImageReply(const QString &imageFileName) {
+QByteArray WbHttpReply::forgeFileReply(const QString &fileName) {
   QByteArray reply;
 
-  QFile imageFile(imageFileName);
-  if (!imageFile.open(QIODevice::ReadOnly))
+  QFile file(fileName);
+  if (!file.open(QIODevice::ReadOnly))
     return forge404Reply();
 
-  QByteArray imageData = imageFile.readAll();
-  int imageSize = imageData.length();
-  QFileInfo fi(imageFile);
-  QString imageExtension = fi.suffix().toLower();
-
+  const QByteArray data = file.readAll();
+  QFileInfo fi(file);
+  const QString extension = fi.suffix().toLower();
```

Comment:
I am not sure about this one. I couldn't find any discussion on this matter, but I believe:
```c++
const QString a = "1";
const QString &b = a;
```
Makes sense (we avoid a copy by using the & operator), but:
```c++
const QString &b = some_function_returning_a_qstring();
```
is equivalent to:
```c++
const QString b = some_function_returning_a_qstring();
```
because in any case the return value of the function has to be stored somewhere.
If both are equivalent, I prefer the second version which is lighter to read.

Reply:
@DavidMansolino found a discussion on this topic: https://stackoverflow.com/questions/4684646/why-assign-a-return-value-to-a-reference

> As the function returns a value you could assign it to a const reference (to a non-const reference would be illegal) and extend the object's lifetime to the lifetime of the reference. However the effect would be exactly the same (const aside) as storing the value in a object directly.

So, it seems the & is indeed not useful in this case.

Reply:
Ok, I couldn't find anything.
So I did some tests and the assignment function is not called independently if the reference is used or not.